### ISSUE TITLE
autoscaler crashes when trying to scale a deployment that has does not exist

### ISSUE THREAD

**Issue by willgraf** [vanvalenlab/kiosk-autoscaler#7]
Hit a weird edge case after deleting the zip deployment (because I was tired of scrolling past it during `get_pods`).  The autoscaler tried to scale the zip consumer because there were not enough pods (0 after deletion), and couldn't find the deployment at all, crashing.

I think it crashed because it is trying to catch an un-imported error?

```
2019-03-15 23:37:11,091 - autoscaler - DEBUG -     Scaling zip-consumer-deployment
Error from server (NotFound): deployments.extensions "zip-consumer-deployment" not found
Traceback (most recent call last):
  File "/bin/autoscale.py", line 78, in _get_kubectl_output
    pods_info = subprocess.check_output(parameter_list)
  File "/usr/lib/python3.6/subprocess.py", line 336, in check_output
    **kwargs).stdout
  File "/usr/lib/python3.6/subprocess.py", line 418, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['kubectl', '-n', 'deepcell', 'describe', 'deployment', 'zip-consumer-deployment']' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/bin/autoscale.py", line 257, in <module>
    autoscaler.redis_monitoring_loop()
  File "/bin/autoscale.py", line 249, in redis_monitoring_loop
    self._scale_deployments()
  File "/bin/autoscale.py", line 220, in _scale_deployments
    deployment)
  File "/bin/autoscale.py", line 148, in _get_current_pods
    deployment_info = self._get_kubectl_output(parameter_list)
  File "/bin/autoscale.py", line 81, in _get_kubectl_output
    except CalledProcessError as err:
NameError: name 'CalledProcessError' is not defined
```

**Comment by willgraf** [comment]
Looks like this error is caused by `CalledProcessError` not being defined.

This is fixed by #11

**Comment by willgraf** [comment]
This is NOT fixed.  Encountered new error:


```
[2019-05-02 21:11:10,527]:[DEBUG]:[Autoscaler]: Deployment `tf-serving` in namespace `deepcell` has a current state of 0 pods and a desired state of 4 pods.
[2019-05-02 21:11:10,539]:[ERROR]:[Autoscaler]: ApiException when calling `patch_namespaced_deployment`: (404)
Reason: Not Found
HTTP response headers: HTTPHeaderDict({'Audit-Id': '686db433-2a22-4eb2-83df-06add3cbf97c', 'Content-Type': 'application/json', 'Date': 'Thu, 02 May 2019 21:11:10 GMT', 'Content-Length': '222'})
HTTP response body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"deployments.apps \"tf-serving\" not found","reason":"NotFound","details":{"name":"tf-serving","group":"apps","kind":"deployments"},"code":404}


[2019-05-02 21:11:10,539]:[CRITICAL]:[autoscale.py]: Fatal Error: ApiException: (404)
Reason: Not Found
HTTP response headers: HTTPHeaderDict({'Audit-Id': '686db433-2a22-4eb2-83df-06add3cbf97c', 'Content-Type': 'application/json', 'Date': 'Thu, 02 May 2019 21:11:10 GMT', 'Content-Length': '222'})
HTTP response body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"deployments.apps \"tf-serving\" not found","reason":"NotFound","details":{"name":"tf-serving","group":"apps","kind":"deployments"},"code":404}
```
---

The API call hits a 404 error with a bad `resource_name` and crashes.  This should handle the 404 gracefully instead.
